src/services/EthService/nodes/web3/node.ts

```diff
@@ -103,3 +119,20 @@ export async function setupWeb3Node() {
     throw new Error('Web3 not found. Please check that MetaMask is installed');
   }
 }
+
+const requestPermission = async (web3Node: Web3Node) => {
+  try {
+    return await web3Node.requestPermissions();
+  } catch (e) {
+    return;
+  }
+};
```

Comment:
Is this necessary? Any reason not to just call `requestPermissions`?

Reply:
The function that calls this uses the `undefined` result to handle this error and fallback to a different web3 access method automatically.